server/map_rpc_server.go

```diff
@@ -310,10 +310,9 @@ func (t *TrillianMapServer) SetLeaves(ctx context.Context, req *trillian.SetMapL
 	ctx, spanEnd := spanFor(ctx, "SetLeaves")
 	defer spanEnd()
 
-	mapID := req.MapId
-	t.setLeafCounter.Add(float64(len(req.Leaves)), string(mapID))
+	t.setLeafCounter.Add(float64(len(req.Leaves)), string(req.MapId))
```

Comment:
This line is weird. I don't think that `string` conversion of `int64` works as intended here.

Reply:
https://play.golang.org/p/SnORcuFf2XU

Reply:
Fixed.

Reply:
Factored out #1898